danskernesdigitalebibliotek/dpl-cms
# Re-factor how to treat and display expired events in frontend. DDFHER-88

### How to test:

Here is an example of an event, that no longer shows the expired events on the list:

https://varnish.pr-1635.dpl-cms.dplplat01.dpl.reload.dk/arrangementer/en-serie-der-bade-har-events-i-fortiden-og-fremtiden

Ontop of that, if you manage to find an expired event, it actually says so on the instance level:

https://varnish.pr-1635.dpl-cms.dplplat01.dpl.reload.dk/arrangementer/en-serie-der-bade-har-events-i-fortiden-og-fremtiden/2024-08-05

<img width="202" alt="Screenshot 2024-10-09 at 08 47 04" src="https://github.com/user-attachments/assets/30d82b24-3f26-444f-910b-93cf2fc54d1e">

Finally, if a series only has one upcoming event (but 1+ expired events), the auto-redirect now redirects to the one active eventinstance.
In the past, you would stay on the series page.

https://varnish.pr-1635.dpl-cms.dplplat01.dpl.reload.dk/arrangementer/series-3-expired-and-1-active-instances

---------

### Suggestions for the reviewer:

1) Review this, commit by commit
2) Make sure to hide whitespace - especially the last commit has a lot of pointless whitespace updates, review-wise :)

<img width="410" alt="Screenshot 2024-10-07 at 13 30 39" src="https://github.com/user-attachments/assets/73a2f121-4548-4de7-88af-f5d6884d86fa">

--------

### Code description:

**Display if expired date on EventInstance full view. DDFHER-88**

This also means that we add the date as a field in the data overview.
This has been cleared with DDF.

**Expose functionality that finds upcoming event IDs. DDFHER-88**

By exposing it as part of the service, we can look up more correctly in
different parts of the code, where we want to see if there are relevant
eventinstances.

**Update EventRedirect code, to only respect upcoming events. DDFHER-88**

We have some code that redirects to the instance, if you visit an
eventseries with only one eventinstance.
However, the existing code only looked at related instances, and not if
the instances are date-expired.
By using the ReoccuringDateFormat service, we can both simplify the
code, and also redirect if there is only one, non-expired instance
available.

**Simplify logic that lists upcoming events. DDFHER-88**

The functionality that filters out expired events from the eventseries
view, can now be removed and replaced by the ReoccuranceDateFormatter
service.
This makes everything much simpler, whilst keeping the code easier to
maintain.

-----

https://reload.atlassian.net/browse/DDFHER-88
